samples/add-feature-collection-layer-from-table/src/main/java/com/esri/arcgismaps/sample/addfeaturecollectionlayerfromtable/components/AddFeatureCollectionLayerFromTableViewModel.kt

```diff
@@ -0,0 +1,237 @@
+                .onSuccess {
+                // Create feature collection tables and add them to the map
+                createAndAddFeatureCollectionLayer()
+            }
+        }
+    }
+
+    private fun createAndAddFeatureCollectionLayer() {
+        try {
+            // Create each feature collection table
+            val pointsTable = createPointsCollectionTable()
+            val linesTable = createLinesCollectionTable()
+            val polygonsTable = createPolygonsCollectionTable()
+
+            // Create a FeatureCollection from the tables
+            val featureCollection = FeatureCollection(featureCollectionTables = listOf(pointsTable, linesTable, polygonsTable))
+
+            // Create a layer from the feature collection and add it to the map
+            val featureCollectionLayer = FeatureCollectionLayer(featureCollection = featureCollection)
+            arcGISMap.operationalLayers.add(featureCollectionLayer)
+        } catch (ex: Exception) {
+            // Surface any unexpected exceptions in a dialog
+            messageDialogVM.showMessageDialog(ex)
+        }
+    }
+
+    /**
+     * Create an in-memory points FeatureCollectionTable, define a simple schema and renderer,
+     * add a single point feature and return the table.
+     */
+    private fun createPointsCollectionTable(): FeatureCollectionTable {
+
+        // Define a simple text field for a place name
+        val placeField = Field(
+            fieldType = FieldType.Text,
+            name = "Place",
+            alias = "Place name",
+            length = 40,
+            isEditable = true,
+            isNullable = false
+        )
+
+        // Create the feature collection table for points using WGS84 spatial reference
+        val pointsCollectionTable = FeatureCollectionTable(
+            fields = listOf(placeField),
+            geometryType = com.arcgismaps.geometry.GeometryType.Point,
+            spatialReference = SpatialReference.wgs84()
+        )
+
+        // Set a simple renderer using a triangle marker symbol
+        pointsCollectionTable.renderer = SimpleRenderer(
+            symbol = SimpleMarkerSymbol(
+                style = SimpleMarkerSymbolStyle.Triangle,
+                color = Color.red,
+                size = 18f
+            )
+        )
+
+        // Create a point geometry (x = longitude, y = latitude) and a feature with attributes
+        val pointGeometry = Point(x = -79.497238, y = 8.849289, spatialReference = SpatialReference.wgs84())
+        val placeAttributes = mapOf("Place" to "Current location")
+
+        // Create and add the feature to the table
+        val pointFeature = pointsCollectionTable.createFeature(placeAttributes, pointGeometry)
+        viewModelScope.launch {
+            pointsCollectionTable.addFeature(pointFeature).onFailure { messageDialogVM.showMessageDialog(it) }
+        }
+
+        return pointsCollectionTable
```

Comment:
This makes it seem like that the `pointsCollectionTable` is returned before the launched scope can add the `pointFeature` to the collection table here. 

I would recommend to convert the `create` functions to be suspend functions which instead returns the `Result<FeatureCollectionTable>` if `addFeature` was successful for points, lines, polygons.

Reply:
made the functions `suspend`